auth/src/main/java/com/firebase/ui/auth/util/FirebaseAuthWrapperImpl.java

```diff
@@ -1,210 +0,0 @@
-
-    public FirebaseAuthWrapperImpl(@NonNull FirebaseAuth firebaseAuth) {
-        if (firebaseAuth == null) {
-            throw new IllegalArgumentException("firebaseAuth must not be null");
-        }
-        mFirebaseAuth = firebaseAuth;
-    }
-
-    @Override
-    @WorkerThread
-    public boolean isExistingAccount(@Nullable final String email) {
-        return email != null && hasProviders(await(mFirebaseAuth.fetchProvidersForEmail(email)));
-    }
-
-    @Override
-    @WorkerThread
-    @NonNull
-    public List<String> getProvidersForEmail(@Nullable String emailAddress) {
-        if (emailAddress == null) {
-            return Collections.emptyList();
-        }
-
-        ProviderQueryResult result =
-                await(mFirebaseAuth.fetchProvidersForEmail(emailAddress));
-        if (hasProviders(result)) {
-            return result.getProviders();
-        }
-
-        return Collections.emptyList();
-    }
-
-    @Override
-    @WorkerThread
-    public boolean resetPasswordForEmail(@NonNull String emailAddress) {
-        Task<Void> curTask = mFirebaseAuth.sendPasswordResetEmail(emailAddress);
-        try {
-            Tasks.await(curTask);
-        } catch (ExecutionException | InterruptedException e) {
-            Log.w(TAG, "attempt to reset password failed", e);
-            return false;
-        }
-        return true;
-    }
-
-    @Override
-    @WorkerThread
-    @Nullable
-    public FirebaseUser signInWithEmailPassword(
-            @NonNull String emailAddress,
-            @NonNull String password) {
-        AuthResult authResult =
-                await(mFirebaseAuth.signInWithEmailAndPassword(emailAddress, password));
-        return authResult == null ? null : authResult.getUser();
-    }
-
-    @Override
-    @WorkerThread
-    @Nullable
-    public FirebaseUser signInWithCredential(@NonNull AuthCredential credential) {
-        Task<AuthResult> curTask = mFirebaseAuth.signInWithCredential(credential);
-        AuthResult authResult = await(curTask);
-        return authResult == null ? null : authResult.getUser();
-    }
-
-    @Override
-    @WorkerThread
-    @Nullable
-    public FirebaseUser getCurrentUser() {
-        return mFirebaseAuth.getCurrentUser();
-    }
-
-    @Override
-    @WorkerThread
-    @Nullable
-    public FirebaseUser createUserWithEmailAndPassword(
-            @NonNull String emailAddress,
-            @NonNull String password) throws ExecutionException, InterruptedException {
-        Task<AuthResult> curTask;
-        curTask = mFirebaseAuth.createUserWithEmailAndPassword(emailAddress, password);
-        Tasks.await(curTask);
-        return curTask.getResult().getUser();
-    }
-
-    @Override
-    public boolean isPlayServicesAvailable(Context context) {
-        return PlayServicesHelper.getInstance(context).isPlayServicesAvailable();
-    }
-
-    @Override
-    public PendingIntent getEmailHintIntent(FragmentActivity fragmentActivity) {
```

Comment:
I'm amazing that nobody noticed: `getEmailHintIntent` was the only method being used in this entire class!!!